.github/workflows/codebuild-ci.yml

```diff
@@ -0,0 +1,46 @@
+name: PR Checks
+on:
+    pull_request:
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+permissions:
+    id-token: write # This is required for requesting the JWT
+
+jobs:
+  codestyle-doc-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.CI_AWS_ROLE_ARN }}
+          aws-region: us-west-2
+          role-duration-seconds: 10800
+      - name: Run Codestyle & Doc Tests
+        uses: aws-actions/aws-codebuild-run-build@v1
+        with:
+          project-name: sagemaker-python-sdk-ci-codestyle-doc-tests
+  unit-tests:
```

Comment:
We will also need timeout settings right? Can we do it here?

Reply:
We are using [`aws-actions/aws-codeubild-run-build@v1` ](https://github.com/marketplace/actions/aws-codebuild-run-build-action-for-github-actions) which allows us to define a bunch of stuff here but doesn't look like timeout settings can be configured here for codebuild projects. Those will still be set in our back infra code

Reply:
Might be good to expose build spec here as well later on, but don't want to do too much at once. Want to get the plan for integ split figured out first before we move on with other improvements

Reply:
Makes sense. let's take small steps